hive/src/main/java/io/confluent/connect/storage/hive/HiveSchemaConverter.java

```diff
@@ -93,4 +131,36 @@ public static TypeInfo convertMap(Schema schema) {
   public static TypeInfo convertPrimitive(Schema schema) {
     return TYPE_TO_TYPEINFO.get(schema.type());
   }
+
+  public static TypeInfo convertPrimitiveMaybeLogical(Schema schema) {
+    if (schema.name() == null) {
+      return convertPrimitive(schema);
+    }
+
+    switch (schema.name()) {
+      case Decimal.LOGICAL_NAME:
+        String scale = schema.parameters().get(Decimal.SCALE_FIELD);
+        String precision = schema.parameters().get(CONNECT_AVRO_DECIMAL_PRECISION_PROP);
+        if (precision == null || Integer.valueOf(precision) <= DECIMAL_PRECISION_DEFAULT) {
+          precision = String.valueOf(DECIMAL_PRECISION_DEFAULT);
+        } else {
+          throw new ConnectException("Illegal precision: "
+              + "Because Hive allows at most 38 precision, the precision"
```

Comment:
```suggestion
              + "Because Hive allows at most DECIMAL_PRECISION_DEFAULT precision, the precision"
```

Reply:
Please accept recommendation but with the correct name. Actually, even better, don't use the string name of the config, but use the config as variable. In that case you have the best of both options. You always have a value in the config, and that value is always the correct one.